textpipe/doc.py

```diff
@@ -608,3 +604,28 @@ def summary(self):
         []
         """
         return self.generate_textrank_summary()
+
+    @functools.lru_cache()
+    def extract_lead(self, n=3):
+        """
+        returns the lead-N sentences (text only) of the document
+        if the text is smaller than the requested N, return full text
+        """
+        return [s[0] for s in self.sents[:n]]
+
+    @property
+    def lead3(self):
```

Comment:
This does not feel very general. Is there a good reason to use `n=3` as some sort of default? Otherwise, maybe we should just do without a property for `lead`? I do like the extensive tests, so keep them if we drop this property.